Feat/update automation stats page - MAILPOET-5700

triple0t opened the pull request:
## Description

We updated the Automation statistics page in this PR.

This PR converts the `Edit automation` button to a split button and adds other buttons.
New buttons:
* Deactivate
* Move to Trash 

<img width="1608" alt="Screenshot 2023-12-04 at 11 32 57 AM" src="https://github.com/mailpoet/mailpoet/assets/30554163/c1b309a8-aae8-4d1d-b192-8c2d4cc2124e">


## Code review notes

Please check commits

## QA notes

* Visit the automation stats page `/wp-admin/admin.php?page=mailpoet-automation-analytics&id=automation_id`
* Ensure the buttons perform the actions intended. 

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5700](https://mailpoet.atlassian.net/browse/MAILPOET-5700)

## After-merge notes

_N/A_

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[MAILPOET-5700]: https://mailpoet.atlassian.net/browse/MAILPOET-5700?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

triple0t committed:
Add split button component to Automation statistics page

MAILPOET-5700

triple0t committed:
Update deactivate button component on Automation statistics page

MAILPOET-5700

triple0t committed:
Update Move to Trash button component on Automation statistics page

MAILPOET-5700

alex-mpoet reviewed (commented):
I found a few issues:
* The button is a bit misaligned in both Firefox and Chrome for me.
Firefox:
![edit-automation-ff](https://github.com/mailpoet/mailpoet/assets/141430865/4c6a1287-4891-4788-9e4e-0c864c3e8d5b)
Chrome:
![edit-automation-chrome](https://github.com/mailpoet/mailpoet/assets/141430865/5c19b9ed-927c-4b58-ba61-02d24769a797)

* After clicking "Deactivate" there is not much visual feedback, the button just disappears, later I noticed a "Draft" badge appearing in the header. But it's sometimes not clear if deactivation happened or not without refreshing.
* How can I see the "Deactivate Now" and "Update & Activate" buttons? After deactivation I only saw "Move to trash".

triple0t commented:
Thank you for the review, @alex-mpoet 👏🏾

> I found a few issues:
> 
> * The button is a bit misaligned in both Firefox and Chrome for me.
>   Firefox:
>   ![edit-automation-ff](https://private-user-images.githubusercontent.com/141430865/288003810-4c6a1287-4891-4788-9e4e-0c864c3e8d5b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NzE4NjYsIm5iZiI6MTcwMTc3MTU2NiwicGF0aCI6Ii8xNDE0MzA4NjUvMjg4MDAzODEwLTRjNmExMjg3LTQ4OTEtNDc4OC05ZTRlLTBjODY0YzNlOGQ1Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwNVQxMDE5MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMTdiMjA0OWYyNWQ2N2I2ZDUyMmJlZGYwZmJlZWFkNDNjMWE3OTZlYWQ0YWI5ZWZkZTY0ZmI4YTEwMTQ1ZWY5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4WRLP8L0Alo0hSwlpjKpWq_-5cIOJ8sw8CAkZYQViFI)
>   Chrome:
>   ![edit-automation-chrome](https://private-user-images.githubusercontent.com/141430865/288003805-5c19b9ed-927c-4b58-ba61-02d24769a797.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NzE4NjYsIm5iZiI6MTcwMTc3MTU2NiwicGF0aCI6Ii8xNDE0MzA4NjUvMjg4MDAzODA1LTVjMTliOWVkLTkyN2MtNGI1OC1iYTYxLTAyZDI0NzY5YTc5Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwNVQxMDE5MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NGQyODkyNTZmM2Y1NWM1MDg1ZDQ3Y2M1NjNiNGQ0ZDVjOTYzODZmY2ZiMTJkMmZmZmM0NzE2YmE0ODgyZDRiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.EI2C3F7Js-MwKEdO-WX0Frur7VAeOzCH11f6pdomo-s)

Noted. 

I'm not seeing the issue on my Chrome and Safari, but I can indeed see the misalignment on Firefox.
I assume it may have something to do with the device screen resolution, width/height, or zoom level.
I'll take a look at this.

> * After clicking "Deactivate" there is not much visual feedback, the button just disappears, later I noticed a "Draft" badge appearing in the header. But it's sometimes not clear if deactivation happened or not without refreshing.

A loading indicator should show
<img width="212" alt="Screenshot 2023-12-05 at 11 58 40 AM" src="https://github.com/mailpoet/mailpoet/assets/30554163/258db1a0-4c90-4347-85f9-b68201e5bafd">

Would you suggest we add a notice here? 
e.g., show a message "Automation deactivated"

> * How can I see the "Deactivate Now" and "Update & Activate" buttons? After deactivation I only saw "Move to trash".

Honestly, I don’t know. 
I assumed you would have more knowledge than I do. I think you worked a little on automation.

I'm reusing the logic available [here](https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/assets/js/src/automation/editor/components/header/index.tsx#L302-L319) as it seems automations can have a `deactivating` state which is separate from `active` and `draft`. 
I believe this happens when automation is in progress, you try to deactivate it, and you select "Let entered subscribers finish the flow" on the prompt.

alex-mpoet commented:
> Would you suggest we add a notice here?
> e.g., show a message "Automation deactivated"

Yes, that would be great.

> I believe this happens when automation is in progress, you try to deactivate it, and you select "Let entered subscribers finish the flow" on the prompt.

Thanks for explaining. For the time being, I just manually set a `deactivating` status in the DB for an automation. There are a couple issues. First, the buttons are stacked on top of each other without any space, I think it would look better with a bit of vertical space between them or if the "Update & Activate" button wasn't blue (`primary` variant).

![image](https://github.com/mailpoet/mailpoet/assets/141430865/168786bd-d243-4895-8797-e9bb835799e4)

Second, the "Update & Activate" button doesn't work, it's disabled.

triple0t committed:
Fix Edit button and icon misalignment

We also removed the ActivateButton because it's not necessary on the Automation analytics pages

MAILPOET-5700

triple0t committed:
Show EditorNotices on Automation analytics pages

MAILPOET-5700

triple0t commented:
Fixed. Thank you for the review, @alex-mpoet 

> Second, the "Update & Activate" button doesn't work, it's disabled.

I've removed the "Update & Activate" button.

Aschepikov commented:
@triple0t I see the issue from Alex's comment with the buttons, is it fixed?
![image](https://github.com/mailpoet/mailpoet/assets/113185437/a502257f-2520-4129-bf66-f637093882ab)

Aschepikov commented:
@triple0t  It depends on the screen width when it is less 837 and from 1114 to 1385 on my chrome

Aschepikov reviewed (approved):
GJ!
